src/public/common/search/Search.js

```diff
@@ -1,113 +1,179 @@
-import React, { Component } from 'react';
-import { withRouter } from 'react-router-dom';
+import React, { useEffect, useState } from 'react';
+import { useLazyQuery } from '@apollo/client';
 import { loader } from 'graphql.macro';
-import AsyncSelect from 'react-select/lib/Async';
-import Control from './Control';
-import Placeholder from './Placeholder';
-import DropdownIndicator from './DropdownIndicator';
-import GroupHeading from './GroupHeading';
-import Option from './Option';
-import ValueContainer from './ValueContainer';
-import { client2 } from '../../client.js';
+import {
+  CircularProgress,
+  makeStyles,
+  InputBase,
+  TextField,
+} from '@material-ui/core';
+import { Autocomplete } from '@material-ui/lab';
+import { Search as SearchIcon, ArrowDropDown } from '@material-ui/icons';
+import { useHistory } from 'react-router-dom';
 
-const SEARCH_QUERY = loader('./SearchQuery.gql');
+import { client2 } from '../../client';
+import useDebounce from '../../../hooks/useDebounce';
+import Option from './Option';
+import Group from './Group';
 
-const groupOptions = (searchData, inputValue) => {
-  return [
-    {
-      value: inputValue,
-      entity: 'search',
-    },
-    {
-      label: 'Top Hit',
-      options: searchData.topHit.hits[0]
-        ? [
-            {
-              ...searchData.topHit.hits[0],
-              entity: 'topHit',
-            },
-          ]
-        : [],
-    },
-    {
-      label: 'Targets',
-      options: searchData.targets.hits,
-    },
-    {
-      label: 'Diseases',
-      options: searchData.diseases.hits,
-    },
-    {
-      label: 'Drugs (Generic Name)',
-      options: searchData.drugs.hits,
-    },
-  ];
-};
+const useStyles = makeStyles(theme => ({
+  listbox: { maxHeight: 'fit-content', color: theme.palette.text.primary },
+  option: { display: 'block', padding: '0 .5rem' },
+  inputBase: {
+    backgroundColor: theme.palette.background.default,
+    color: theme.palette.text.secondary,
+    width: '100%',
+  },
+  inputBaseInput: { padding: '.25rem .5rem' },
+  root: embedded => ({ width: embedded ? '400px' : '', marginRight: '2rem' }),
+}));
 
-const IndicatorSeparator = () => null;
+function Search({ autoFocus = false, embedded = false }) {
+  const [open, setOpen] = React.useState(false);
+  const [inputValue, setInputValue] = useState('');
+  const debouncedInputValue = useDebounce(inputValue, 300);
+  const SEARCH_QUERY = loader('./SearchQuery.gql');
```

Comment:
Can you move this line outside of the component? It does not depend on props or state and we don't want it to execute for every render.

Reply:
Makes sense!